### PR TITLE
Handle info, group, and groupCollapsed in Strict Mode logging

### PR DESCRIPTION
**Pull request by timneutkens**
## Summary

While working on the new Next.js router which heavily relies on useReducer I noticed that `group` and `groupCollapsed` which both take labels were showing as-is in the console for the second render/dispatch in Strict Mode logs. While looking at the code I found that `info` was also not instrumented.

I've added additional handling for:
- `info`
- `group`
- `groupCollapsed`

## How did you test this change?

Looked at earlier PRs and copied the blocks for the `error` case in the same file to cover the other console methods.

Somehow the tests are failing while running locally but I'm not sure if that's caused by my local build because when I make the change in the extension script in devtools it does show correctly.

Example of the change. `prefetch` is before, `navigate` is after:

Screen Shot 2022-09-01 at 17 40 36

### CODE REVIEW

**Comment by gaearon** [comment]
thanks! some test is unhappy.

**Comment by timneutkens** [comment]
Yeah it's the same failure as local... will investigate it further 😓

File: packages/react-devtools-shared/src/__tests__/console-test.js [hunk 1]
```diff
@@ -528,6 +540,36 @@ describe('console', () => {
...
       'error',
     ]);
+
+    expect(mockInfo).toHaveBeenCalledTimes(2);
+    expect(mockInfo.mock.calls[0]).toHaveLength(1);
+    expect(mockInfo.mock.calls[0][0]).toBe('info');
+    expect(mockInfo.mock.calls[1]).toHaveLength(3);
+    expect(mockInfo.mock.calls[1]).toEqual([
+      '%c%s',
+      `color: ${process.env.DARK_MODE_DIMMED_ERROR_COLOR}`,
```
Comment (gnoff) [inline packages/react-devtools-shared/src/__tests__/console-test.js:550]: This probabaly need to just be LOG_COLOR not ERROR_COLOR

**Review by gnoff — CHANGES_REQUESTED** [review]

File: packages/react-devtools-shared/src/__tests__/console-test.js (hunk 1 above)
Comment (timneutkens) [reply packages/react-devtools-shared/src/__tests__/console-test.js:550]: Ah missed that these differ!

**Comment by timneutkens** [comment]
Should be good to land now 👍

**Review by gnoff — APPROVED** [review]

**Comment by mondaychen** [comment]
If we are overriding `group` and `groupCollapsed`, shall we do the same for `groupEnd` to keep the format correct? (Especially when your are using multiple levels of grouping).
Also how about other output related methods such as `console.table`?

**Comment by timneutkens** [comment]
> If we are overriding group and groupCollapsed, shall we do the same for groupEnd to keep the format correct? (Especially when your are using multiple levels of grouping).

I only added `group` and `groupCollapsed` as these take a `label` argument whereas `groupEnd` does not take parameters so overriding doesn't add any additional handling: https://developer.mozilla.org/en-US/docs/Web/API/console/groupEnd

> Also how about other output related methods such as console.table?

`console.table` takes in an object/array which is currently passed through as-is in the handling. I couldn't find if it supports `%c` (styling of console messages) and from trying it doesn't seem like it is supported unfortunately.

**Review by mondaychen — APPROVED** [review]
lgtm

**Pull request merged**